src/main/java/bit/javaoop/SalaryCalculator.java

```diff
@@ -0,0 +1,5 @@
+package bit.javaoop;
+
+public interface SalaryCalculator {
```

Comment:
Tutaj parę słów bo już wyskoczyłeś w interfejsy  i zrobiło się lekkie spaghetti :), generalnie słówka kluczowego **implements** używamy do wyrażania, że coś **posiada** jakieś cechy np. Manager implements Employee można by wyrazić w sposób taki, że manager posiada cechy pracownika a jeszcze bardziej dosłownie było by np. Bird implements CanFly.

Chcąc naprawdę bardzo wykorzystać tą klasę SalaryCalculator to tak średnio brzmi, że człowiek posiada cechy kalkulatora choć może i ma sens ja bym raczej wkomponował SalaryCalculator jako pole klas Managerów wtedy nam się sprawa trochę komplikuje.
Musimy stworzyć jakieś implementacje _kalkulatorów_ i później np. konkretną implementację przyjmować w konstruktorze bądź poprostu inicjalizować nasze pole konkretną implementacją w konstruktorze.
Chcelibyśmy dojść do takiego punktu gdzie metoda getSalary w naszych managerach wygląda tak : 
```
public Float getSalary() {
        return salaryCalculator.calculate();
    }
```
Biorąc to wszystko pod uwagę po prostu w tym wypadku odpuściłbym sobie SalaryCalculator 😄 

Właściwie jest to doskonały przykład złamania zasad YAGNI,KISS i DRY
- You Aren't Gonna Need It ( Zwracanie własnego pola salary nie jest tak skomplikowane żeby była do tego potrzebna osobna klasa)
- Keep It Simple Stupid ( Nadmierne komplikowanie kodu zrobiło sie spaghetti )
- Don't Repeat Yourself  ( Mamy już metodę getSalary w AbstractEmployee ) 😄